snmp/tests/test_e2e_core_profiles/test_profile_riverbed_interceptor.py

```diff
@@ -0,0 +1,118 @@
+
+    # --- TEST METRICS ---
+    assert_common_metrics(aggregator, common_tags)
+
+    aggregator.assert_metric('snmp.cpu.usage', metric_type=aggregator.GAUGE, tags=common_tags)
+    aggregator.assert_metric('snmp.memory.total', metric_type=aggregator.GAUGE, tags=common_tags)
+    aggregator.assert_metric('snmp.memory.usage', metric_type=aggregator.GAUGE, tags=common_tags)
+    aggregator.assert_metric('snmp.memory.used', metric_type=aggregator.GAUGE, tags=common_tags)
+    # aggregator.assert_metric(
+    #     'snmp.reverbed.interceptor.serviceStatus', metric_type=aggregator.GAUGE, tags=common_tags
+    # )
+    tag_rows = [
+        [
+            'reverbed_interceptor_proc_name:but acted Jaded but zombies their but',
+            'reverbed_interceptor_proc_status:acted kept their Jaded Jaded driving Jaded acted',
+        ],
+        [
+            'reverbed_interceptor_proc_name:forward zombies',
+            'reverbed_interceptor_proc_status:their driving oxen acted oxen but acted',
+        ],
+        [
+            'reverbed_interceptor_proc_name:oxen their oxen acted quaintly their oxen',
+            'reverbed_interceptor_proc_status:driving quaintly zombies but',
+        ],
+        [
+            'reverbed_interceptor_proc_name:quaintly quaintly acted kept',
+            'reverbed_interceptor_proc_status:acted driving acted Jaded oxen',
+        ],
+    ]
+    for tag_row in tag_rows:
+        aggregator.assert_metric(
+            'snmp.reverbed.interceptor.proc', metric_type=aggregator.GAUGE, tags=common_tags + tag_row
+        )
+
+    tag_rows = [
+        ['reverbed_interceptor_neighbor_name:acted zombies their quaintly Jaded forward'],
+        ['reverbed_interceptor_neighbor_name:driving'],
+        ['reverbed_interceptor_neighbor_name:forward acted quaintly but oxen oxen their acted Jaded'],
+        ['reverbed_interceptor_neighbor_name:oxen acted but'],
+    ]
+    for tag_row in tag_rows:
+        aggregator.assert_metric(
+            'snmp.reverbed.interceptor.neighborConnectionCount',
+            metric_type=aggregator.GAUGE,
+            tags=common_tags + tag_row,
+        )
+
+    aggregator.assert_metric('snmp.ifBandwidthInUsage.rate')
+    aggregator.assert_metric('snmp.ifBandwidthOutUsage.rate')
+    aggregator.assert_metric('snmp.ifHCInBroadcastPkts')
+    aggregator.assert_metric('snmp.ifHCInMulticastPkts')
+    aggregator.assert_metric('snmp.ifHCInOctets')
+    aggregator.assert_metric('snmp.ifHCInOctets.rate')
+    aggregator.assert_metric('snmp.ifHCInUcastPkts')
+    aggregator.assert_metric('snmp.ifHCOutBroadcastPkts')
+    aggregator.assert_metric('snmp.ifHCOutMulticastPkts')
+    aggregator.assert_metric('snmp.ifHCOutOctets')
+    aggregator.assert_metric('snmp.ifHCOutOctets.rate')
+    aggregator.assert_metric('snmp.ifHCOutUcastPkts')
+    aggregator.assert_metric('snmp.ifHighSpeed')
+    aggregator.assert_metric('snmp.ifInSpeed')
+    aggregator.assert_metric('snmp.ifOutSpeed')
```

Comment:
I'm also a bit confused about those assertions (they were added manually), is there any other way to handle these metrics?

Reply:
you have these assertions because these metrics are present in your snmprec
you don't need to assert all of this since we already have the `assert_extend_generic_if`
since these are not specific to this profile there's no need to keep them, what you can do in this case is remove the entries from your snmprec
they should be the lines with the OIDs `1.3.6.1.2.1.31.1.*`